cli/azd/pkg/input/console.go

```diff
@@ -647,6 +647,19 @@ func watchConsoleWidth(c *AskerConsole) {
 	}
 }
 
+func watchTerminalInterrupt(c *AskerConsole) {
+	signalChan := make(chan os.Signal, 1)
+	signal.Notify(signalChan, os.Interrupt)
+	go func() {
+		<-signalChan
+
+		// unhide the cursor if applicable
+		_ = c.spinner.Stop()
+
+		os.Exit(1)
```

Comment:
This is definitely a non-graceful termination similar to [gh](https://github.com/cli/cli/blob/3bb62d4724b324bb5d6f0395aca93c9b9834cab8/pkg/iostreams/iostreams.go#L330), but I don't think we need to support cleanups right now.

Reply:
Spoke to soon as I think about #2810

Reply:
Happy if you want to defer the cancellation half for now, but also excited that you are thinking about it :-).  I suspect supporting Ctrl-C doing graceful shutdown of `azd` (including aborting pending inflight operations when it makes sense) will end up bloating this PR, so happy for you to land this quick fix and look at cancellation end to end later if you feel that's better.

I think we have a good start in supporting this since we are threading context everywhere, but we still need to understand where to poll for cancellation and then how to safely shutdown.  If it looks like it's starting to become a ton of work, feel free to push it to another PR.

Going to be exciting :-)

Reply:
Thanks for all the good feedback.

Just sharing some early thoughts: I was thinking if we wanted to support cancellation, we'd actually introduce a static registration of handlers. Specific parts of the codebase could register cancellation handlers and gracefully terminate, then deregister once out-of-scope.

However, by default, in most of the codebase, an interrupt is an immediate shutdown. The main reason I was thinking along these lines is because the default behavior is more correct -- users expect fast termination from a CLI application. And most CLI operations don't really require any real type of resource cleanup that would otherwise cause leaks that prevail past the process context. Implementing bug-free cancellation in all parts of the codebase is also non-trivial and requires constant testing.

Reply:
Going to punt this down the road, but hopefully the context here helps